tests/Microsoft.Bot.Expressions.Tests/ExpressionEngineTests.cs

```diff
@@ -744,6 +745,42 @@ public void EvaluateJson(string input, object expected, HashSet<string> expected
             }
         }
 
+        [TestMethod]
+        public void TestAccumulatePath()
+        {
+            var memory = new SimpleObjectMemory(new
+            {
+                f = "foo",
+                b = "bar",
+                z = new
+                {
+                    z = "zar"
+                }
+            });
+
+            var parser = new ExpressionEngine();
+
+            // normal case, note, we doesn't append a " yet
+            var exp = parser.Parse("a[f].b[b].z");
```

Comment:
Here i tested the `TryAccumulatePath` directly, it supports
* `a[b].c`
* `a[b][c].d`